core/txpool/txpool.go

```diff
@@ -91,10 +91,18 @@ func (p *TxPool) Close() error {
 	errs = append(errs, <-errc)
 
 	// Terminate each subpool
+	var anyErr bool
 	for _, subpool := range p.subpools {
-		errs = append(errs, subpool.Close())
+		err := subpool.Close()
+		if err != nil {
+			anyErr = true
+			errs = append(errs, err)
+		}
+	}
+	if anyErr {
+		return fmt.Errorf("subpool close errors: %v", errs)
 	}
-	return errors.Join(errs...)
+	return nil
 }
```

Comment:
You are losing/ignoring the error from line 91.

Reply:
My 2c is that anyErr seems very convoluted, just check for len(errs) == 0 (and omit pushing nil into it in line 91).